### PR TITLE
Missing basePlugin types (#9175)

### PR DESCRIPTION
**Pull request by mjastrzebowski**
### Context

Some properties were missing from the typescript definition of the BasePlugin class 

### How has this been tested?

Tested manually typescript definitions 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. https://github.com/handsontable/handsontable/issues/9175

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: handsontable/types/plugins/base/base.d.ts [hunk 1]
```diff
@@ -17,6 +20,7 @@ export class BasePlugin {
   disablePlugin(): void;
   updatePlugin(): void;
   addHook(name: string, callback: () => void): void;
+  addHook<K extends keyof Events>(key: K, callback: Events[K] | Array<Events[K]>): void;
   removeHooks(name: string): void;
```
Comment (budnix) [inline handsontable/types/plugins/base/base.d.ts:24]:
```suggestion
  addHook<K extends keyof Events>(key: K, callback: Events[K] | Array<Events[K]>): void;
  removeHooks<K extends keyof Events>(name: K): void;
```
I think that the old typing for `addHook` can be replaced with that new one completely. WDYT to fix typing for the `removeHooks` method right away? Underneath the method uses the `removeHook` from the Core that that have [similar types](https://github.com/handsontable/handsontable/blob/master/handsontable/types/core.d.ts#L114).

**Review by budnix — CHANGES_REQUESTED** [review]
It looks good. Besides my comments with improvement suggestions, I would ask you to update the tests that check that types.

File: `handsontable/src/plugins/base/__tests__/base.types.ts`
```diff
import Handsontable from 'handsontable';
import { BasePlugin } from 'handsontable/plugins';

const elem = document.createElement('div');
const hot = new Handsontable(elem, {});

const pluginKey: string = BasePlugin.PLUGIN_KEY;
const settingKeys: string[] | boolean = BasePlugin.SETTING_KEYS;

const basePlugin = new BasePlugin(hot);

-
+const hotPlugin: Handsontable = basePlugin.hot;

basePlugin.init();
basePlugin.enablePlugin();
basePlugin.disablePlugin();
-basePlugin.addHook('name', () => {});
+basePlugin.addHook('beforeKeyDown', (event: any) => { return false });
-basePlugin.removeHooks('name');
+basePlugin.removeHooks('beforeKeyDown');
basePlugin.clearHooks();
basePlugin.callOnPluginsReady(() => {});
basePlugin.destroy();
```
⬆️  WDYT?

File: handsontable/types/plugins/base/base.d.ts (hunk 1 above)
Comment (mjastrzebowski) [reply handsontable/types/plugins/base/base.d.ts:24]: I wanted to replace it, idk why I left the old one :)

**Review by budnix — APPROVED** [review]
Thanks! 🏅

**Pull request merged**